ext/pgsql/php_pgsql.h

```diff
@@ -184,7 +184,7 @@ static const php_stream_ops php_stream_pgsql_fd_ops = {
 ZEND_BEGIN_MODULE_GLOBALS(pgsql)
 	zend_long num_links,num_persistent;
 	zend_long max_links,max_persistent;
-	zend_long allow_persistent;
+	bool allow_persistent;
 	zend_long auto_reset_persistent;
 	int ignore_notices,log_notices;
```

Comment:
@nielsdos wouldn't that change leave some unused bytes in the struct? Should the field be moved to the end?

Reply:
There's a 7 byte hole after allow_persistent with this change. Moving it to the end won't fill the hole.
I generally also don't try to move too much in the struct to keep all the relevant fields close to each other in the same cache line.
You can fit an int in there though, so you could do:
```c
	zend_long num_links,num_persistent;
	zend_long max_links,max_persistent;
	zend_long allow_persistent;
	bool allow_persistent;
    int ignore_notices;
	zend_long auto_reset_persistent;
	int log_notices;
```
This is the best you can do.

Reply:
ideally manual padding is better I think but sure I can always do what you mention.